sql/catalyst/src/test/scala/org/apache/spark/sql/util/TimestampFormatterSuite.scala

```diff
@@ -44,7 +44,8 @@ class TimestampFormatterSuite extends SparkFunSuite with SQLHelper with Matchers
     DateTimeTestUtils.outstandingTimezonesIds.foreach { zoneId =>
       val formatter = TimestampFormatter(
         "yyyy-MM-dd'T'HH:mm:ss.SSSSSS",
-        DateTimeUtils.getZoneId(zoneId))
+        DateTimeUtils.getZoneId(zoneId),
+        needVarLengthSecondFraction = true)
```

Comment:
is this really needed? The test input is fixed: `2018-12-02T10:11:12.001234`

Reply:
We are not going to use fixed-length formatter for parsing right?  So I guess test that one is not needed anymore. the var-length formatter should work fine with fixed inputs.